### DISCUSSION TITLE
Initial stresss and boundary condition implementation - PorousFlow

### DISCUSSION THREAD

**Discussion by anupkgeos** [idaholab/moose#26104] [Q&A Modules: Porous Flow] [answered]
Hi,
Considering the following scenario (Top and RIght BCs),
![image](https://github.com/idaholab/moose/assets/145852108/2f17147e-5798-4d27-bb29-54a472d9eb5b)
![image](https://github.com/idaholab/moose/assets/145852108/efb86620-4ee7-4341-be10-0b5924c8756c)

1) how to implement a free-to-move boundary on top?  2)  initial stress and right boundary for displacement variable are implemented as:

Initial Stress Condition :

```
Material:
   [ini_strain]
     type = ComputeEigenstrainFromInitialStress
     initial_stress = 'sxx0 0 0  0 syy0 0  0 0 0'
     eigenstrain_name = 'ini_strain'
   []
```
Boundary: 
```
  [right_u]
   type = Pressure
   boundary = right
   variable = disp_x
   component = 0
   function = '0.7*2260*10*y'  # sigma_h = 0.7*sigma_v (total stress, compressive)
   use_displaced_mesh = false
 []
```

Functions:
```
gravity = -9.81
solid_density = 2260
fluid_density = 1059
porosity0 = 0.1
[Functions]
  [p0]
    type = ParsedFunction
    expression = '0.1e6 - 9.81e3 * y' # -ve y 9.81 MPa/km = 9.81e6/1000m = 9.81e3Pa/m
    execute_on = INITIAL
  []
  [sxx0]
    type = ParsedFunction
    expression = '-0.7*${gravity} * y * (${solid_density} - ${fluid_density}) * (1.0 - ${porosity0})'  # initial effective stress that should result from weight force
  []
  [syy0]
    type = ParsedFunction
    expression = '-${gravity} * y * (${solid_density} - ${fluid_density}) * (1.0 - ${porosity0})'  # initial effective stress that should result from weight force
  []
[]
```
with this implementation, I am getting a very high displacement value. Can someone please help me with what is wrong here? thanks!

**Comment by josebastiase** [comment, marked as answer]
Hi,
For me at least, is easier if you link a minimal example.

You can set a free displacement boundary just not setting any mechanical condition in that boundary.

I would set the load at the boundaries as total stress
```
  [load]
    type = FunctionNeumannBC
    variable = disp_x
    function = whatever
    boundary = right
  []
```

hope it helps...
